File: backend/app/schemas/project.py
from pydantic import BaseModel, Field, HttpUrl
from typing import Optional, List
from datetime import datetime


class ProjectCreate(BaseModel):
    """项目创建模型"""
    name: str = Field(..., min_length=1, max_length=100, description="项目名称")
    description: str = Field(..., min_length=1, max_length=1000, description="项目描述")
    tech_stack: List[str] = Field(..., min_items=1, description="技术栈")
    cover_image: Optional[str] = Field(None, max_length=500, description="封面图片 URL")
    demo_url: Optional[str] = Field(None, max_length=500, description="演示地址")
    github_url: Optional[str] = Field(None, max_length=500, description="GitHub 地址")
    start_date: Optional[datetime] = Field(None, description="开始日期")
    end_date: Optional[datetime] = Field(None, description="结束日期")
    status: str = Field(default="completed", description="项目状态: completed/in_progress/planned")
    highlights: Optional[List[str]] = Field(None, description="项目亮点")
    order: int = Field(default=0, description="排序顺序")


class ProjectUpdate(BaseModel):
    """项目更新模型"""
    name: Optional[str] = Field(None, min_length=1, max_length=100)
    description: Optional[str] = Field(None, min_length=1, max_length=1000)
    tech_stack: Optional[List[str]] = Field(None, min_items=1)
    cover_image: Optional[str] = Field(None, max_length=500)
    demo_url: Optional[str] = Field(None, max_length=500)
    github_url: Optional[str] = Field(None, max_length=500)
    start_date: Optional[datetime] = None
    end_date: Optional[datetime] = None
    status: Optional[str] = None
    highlights: Optional[List[str]] = None
    order: Optional[int] = None


class ProjectResponse(BaseModel):
    """项目响应模型"""
    id: str
    name: str
    description: str
    tech_stack: List[str]
    cover_image: Optional[str]
    demo_url: Optional[str]
    github_url: Optional[str]
    start_date: Optional[datetime]
    end_date: Optional[datetime]
    status: str
    highlights: Optional[List[str]]
    order: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/api/project.ts
import request from '@/utils/request'

export interface Project {
  id: string
  name: string
  description: string
  tech_stack: string[]
  cover_image?: string
  demo_url?: string
  github_url?: string
  start_date?: string
  end_date?: string
  status: 'completed' | 'in_progress' | 'planned'
  highlights?: string[]
  order: number
  created_at: string
  updated_at: string
}

export interface ProjectCreate {
  name: string
  description: string
  tech_stack: string[]
  cover_image?: string
  demo_url?: string
  github_url?: string
  start_date?: string
  end_date?: string
  status?: string
  highlights?: string[]
  order?: number
}

export interface ProjectUpdate {
  name?: string
  description?: string
  tech_stack?: string[]
  cover_image?: string
  demo_url?: string
  github_url?: string
  start_date?: string
  end_date?: string
  status?: string
  highlights?: string[]
  order?: number
}

/**
 * 获取项目列表
 */
export const getProjects = (params?: {
  status?: string
  tech?: string
  skip?: number
  limit?: number
}) => {
  return request({
    url: '/projects',
    method: 'get',
    params
  })
}

/**
 * 获取精选项目
 */
export const getFeaturedProjects = (limit: number = 6) => {
  return request({
    url: '/projects/featured',
    method: 'get',
    params: { limit }
  })
}

/**
 * 获取技术栈列表
 */
export const getTechStack = () => {
  return request({
    url: '/projects/tech-stack',
    method: 'get'
  })
}

/**
 * 获取单个项目
 */
export const getProject = (id: string) => {
  return request({
    url: `/projects/${id}`,
    method: 'get'
  })
}

/**
 * 创建项目
 */
export const createProject = (data: ProjectCreate) => {
  return request({
    url: '/projects',
    method: 'post',
    data
  })
}

/**
 * 更新项目
 */
export const updateProject = (id: string, data: ProjectUpdate) => {
  return request({
    url: `/projects/${id}`,
    method: 'put',
    data
  })
}

/**
 * 删除项目
 */
export const deleteProject = (id: string) => {
  return request({
    url: `/projects/${id}`,
    method: 'delete'
  })
}
